### PR TITLE
Add ut for querycoord's cluster

### PR DESCRIPTION
**Pull request by xige-16**
fix issue: #7192
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/querycoord/querynode.go [hunk 1]
```diff
@@ -564,10 +586,10 @@ func (qn *queryNode) loadSegments(ctx context.Context, in *querypb.LoadSegmentsR
 }
 
 func (qn *queryNode) releaseSegments(ctx context.Context, in *querypb.ReleaseSegmentsRequest) error {
-	qn.serviceLock.RLock()
-	onService := qn.onService
-	qn.serviceLock.RUnlock()
-	if !onService {
+	qn.stateLock.RLock()
```
Comment (czs007) [inline internal/querycoord/querynode.go:589]: why not add a function getState()?

File: internal/querycoord/querynode.go [hunk 2]
```diff
@@ -564,10 +586,10 @@ func (qn *queryNode) loadSegments(ctx context.Context, in *querypb.LoadSegmentsR
...
+	state := qn.state
+	qn.stateLock.RUnlock()
+	if state != online {
```
Comment (czs007) [inline internal/querycoord/querynode.go:592]: why not add a function named getState() or isOnline()?

Comment (xige-16) [reply internal/querycoord/querynode.go:592]: Isn't it the same？

Comment (czs007) [reply internal/querycoord/querynode.go:592]: Simplify the code

**Pull request merged**